sdk/core/Azure.Core/src/RequestOptions.cs

```diff
@@ -0,0 +1,144 @@
+        {
+            Policies ??= new();
+            Policies.Add((position, policy));
+        }
+
+        /// <summary>
+        /// Customizes the <see cref="ResponseClassifier"/> for this operation.
+        /// Adding a classifier changes the default classification behavior so that it considers
+        /// the passed-in status code to be an error or not, as specified.
+        /// This is useful for cases where you'd like to prevent specific response status codes from being treated as errors by
+        /// logging and distributed tracing policies -- that is, if a response is not classified as an error, it will not appear as an error in
+        /// logs or distributed traces.
+        /// </summary>
+        /// <param name="statusCode">The status code to customize classification for.</param>
+        /// <param name="isError">Whether the passed-in status code should be classified as an error.</param>
+        public void AddClassifier(int statusCode, bool isError)
+        {
+            Argument.AssertInRange(statusCode, 100, 599, nameof(statusCode));
+
+            if (_frozen)
+            {
+                throw new InvalidOperationException("Cannot modify this RequestContext after it has been used in a method call.");
+            }
+
+            int length = _statusCodes == null ? 0 : _statusCodes.Length;
+            Array.Resize(ref _statusCodes, length + 1);
+            _statusCodes[length] = (statusCode, isError);
+        }
+
+        /// <summary>
+        /// Customizes the <see cref="ResponseClassifier"/> for this operation.
+        /// Adding a <see cref="ResponseClassificationHandler"/> changes the classification
+        /// behavior so that it first tries to classify a response via the handler, and if
+        /// the handler doesn't have an opinion, it instead uses the default classifier.
+        /// Handlers are applied in order so the most recently added takes precedence.
+        /// This is useful for cases where you'd like to prevent specific response status codes from being treated as errors by
+        /// logging and distributed tracing policies -- that is, if a response is not classified as an error, it will not appear as an error in
+        /// logs or distributed traces.
+        /// </summary>
+        /// <param name="classifier">The custom classifier.</param>
+        public void AddClassifier(ResponseClassificationHandler classifier)
+        {
+            if (_frozen)
+            {
+                throw new InvalidOperationException("Cannot modify this RequestContext after it has been used in a method call.");
+            }
+
+            int length = _handlers == null ? 0 : _handlers.Length;
+            Array.Resize(ref _handlers, length + 1);
+            Array.Copy(_handlers, 0, _handlers, 1, length);
+            _handlers[0] = classifier;
+        }
+
+        internal void Freeze()
+        {
+            _frozen = true;
+        }
+
+        /// <summary>
+        /// </summary>
```

Comment:
populate or remove?

Reply:
I need an automated check to flag these - I do it almost every time, arg.  🤦‍♀️